src/QsCompiler/Tests.Compiler/SerializationTests.fs

```diff
@@ -261,7 +260,8 @@ module SerializationTests =
         let specs = attrs.Specializations |> Seq.map (fun s -> (s.ToTuple() |> fst).ToJson()) |> Seq.toList
         let AssertEqual (expected : string list) (got : _ list) = 
             Assert.Equal(expected.Length, got.Length)
-            expected |> List.iteri (fun i ex -> Assert.Equal (ex.Replace("%%%", dllId.Value), got.[i]))
+            expected |> List.iteri (fun i ex ->
+                Assert.Equal (ex.Replace("%%%", HttpUtility.JavaScriptStringEncode dllId.Value), got.[i]))
```

Comment:
Why is the java script string encoding needed here?

Reply:
The `%%%` markers occur inside of a JSON-serialized syntax tree. Some characters in the source file path need to be escaped (for example `\` should become `\\`).